base/math.jl

```diff
@@ -488,10 +492,12 @@ asinh(x::Number)
 
 # functions that return NaN on non-NaN argument for domain error
 """
-    sin(x)
+    sin(x::T) where T <: Number -> float(T)
 
 Compute sine of `x`, where `x` is in radians.
 
+Throws a [`DomainError`](@ref) if `isinf(x)`, returns a `T(NaN)` if `isnan(x)`.
```

Comment:
According to the [doc style guide](https://docs.julialang.org/en/v1/manual/documentation/), it should be imperative ("throw", "return").